app/code/Magento/Cron/Model/ResourceModel/Schedule/ExpressionValidator.php

```diff
@@ -0,0 +1,213 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Cron\Model\ResourceModel\Schedule;
+
+use Magento\Cron\Model\Schedule;
+use Magento\Framework\Exception\CronException;
+
+/**
+ * Schedule cron expression validator
+ *
+ * @api
+ */
+class ExpressionValidator
+{
+    const CRON_EXPRESSION_MIN_PARTS_NUMBER = 5;
+    const CRON_EXPRESSION_MAX_PARTS_NUMBER = 6;
+
+    /**
+     * @var Schedule
+     */
+    private $schedule;
+
+    /**
+     * @var string
+     */
+    private $lastErrorMsg;
+
+    /**
+     * @var array
+     */
+    private $fromToModulus;
+
+    /**
+     * @param Schedule $schedule
+     */
+    public function setSchedule($schedule)
+    {
+        $this->schedule = $schedule;
+    }
+
+    /**
+     * @return string
+     */
+    public function getLastErrorMsg()
+    {
+        return $this->lastErrorMsg;
+    }
+
+    /**
+     * Checks the schedule's cron expression validity
+     *
+     * @return bool
+     */
+    public function checkCronExpr()
+    {
+        $expr = $this->schedule->getCronExprArr();
+
+        if (!$expr) {
+            return false;
+        }
+
+        if (sizeof($expr) < self::CRON_EXPRESSION_MIN_PARTS_NUMBER
```

Comment:
Usage of `count` is more conventional. Simplify task for these who will read code after.

Reply:
I thought the same when I saw this at the original code in app/code/Magento/Cron/Model/Schedule.php, it uses `sizeof` instead of `count`. I tried to preserve it, but I will change it.